freqtrade/__init__.py

```diff
@@ -1,5 +1,15 @@
 """ FreqTrade bot """
-__version__ = '2019.7-dev'
+__version__ = 'develop'
+
+if __version__ == 'develop':
+
+    try:
+        import subprocess
```

Comment:
This is currently called on every startup for develop.
Not sure if that's a problem (timing-vise) or if we can assume it's ok.

Reply:
Performance impact is not visible, as far as I see...

* The output is strange: `freqtrade b'0.14.3-5523-ga7e45c5a'`
* It does not work if a user runs the installed version from another dir.
* This will not be needed if freqtrade is installed via pupy. Is it possible to publish the develop asset to pupy (when this will be implemented for master) after each merge to develop with corresponding git commit id inside?

Reply:
yes, the output can be improved by using a different git command (probably `git log ...` which shows the short version of the hash).

> It does not work if a user runs the installed version from another dir.

Yes - then it'll simply print `develop` - which is similar to the current output (`2019-7-dev` - which does not say much more about the version.

For master, we won't need this, since we'll increment the version as we do now.

I doubt that we'll push develop versions to pypi ... while it's probably possible, it'll create a mess on what the current version is (pypi does not support different branches afaik) - we would have to either publish to packages, or play with version-numbers to make them unique (and always select the correct one ...) which i think is difficult.

I don't think once we're on pypi too many people will be using develop other than developers (and people living on the edge). I guess we'll then have it somewhat outomated so we can release more often (although surely not at the frequency of ccxt with ~4 releases per day ...)